calico_node/startup/startup.go

```diff
@@ -128,6 +130,31 @@ func main() {
 	message("Using node name: %s", nodeName)
 }
 
+// configureNodeRef will attempt to discover the cluster type it is running on, check to ensure we
+// have not already set it on this Node, and set it if need be.
+func configureNodeRef(node *api.Node) {
+	orchestrator := ""
+	nodeRef := ""
+	// Sort out what type of cluster we're running on.
+	if nodeRef = os.Getenv("CALICO_K8S_NODE_REF"); nodeRef != "" {
```

Comment:
This could be simplified, since we only support this feature on k8s right now.

```go
orch := "k8s"
nodeRef := ""
if nodeRef = os.Getenv("CALICO_K8S_NODE_REF"); nodeRef == "" {
  // No reference specified, return.
  return
}
```

Reply:
yea, in my mind I was thinking "We will probably want to check for other orchestrator env vars." but I don't know what we plan to set for their env vars or how imminent their implementation will be.

Reply:
They're not planned any time soon, so I'd opt for the simpler approach here.

We'll cross that bridge when we get to it - generally best to keep things as simple as possible until required.